effects/butterfly-gears/textureloop.s

```diff
@@ -0,0 +1,27 @@
+; vim: ft=asm68k:ts=8:sw=8:
+
+ROTZOOM_W			= 24
+ROTZOOM_H			= 24
+COPPER_HALFROW_INSTRUCTIONS	= ROTZOOM_W/2+2
+SINGLEROW_SIZE			= COPPER_HALFROW_INSTRUCTIONS*4
+
+	xdef _PlotTextureAsm
+
+	section	'.text',code
+
+; -----------------------------------------------------------------------------
+; _PlotTextureAsm(copperDst, texture, uv, uvDeltaRow, uvDeltaCol)
+;                 a0         a1       d5  d6          d1
+; -----------------------------------------------------------------------------
+
+_PlotTextureAsm:
+	movem.l	d2-d7/a2-a6,-(a7)
+	pea	restore
```

Comment:
Interesting... is it really necessary to push `restore` address on stack and use `rts` later? Could I just remove those two lines and it'd work the same way?

Reply:
Not really. The original asm code hat multiple exit paths initially, this was to make sure, the register restore happens with every `rts`. Left it for now, might change later